app/controllers/checkouts_controller.rb

```diff
@@ -0,0 +1,95 @@
+class CheckoutsController < ApplicationController
+  rescue_from Pundit::NotAuthorizedError, with: :user_not_authorized, except: :update
```

Comment:
Will remove this. This is because I moved the `update` action to `PurchasesController` early.